flytepropeller/pkg/controller/nodes/array/handler.go

```diff
@@ -192,11 +192,20 @@ func (a *arrayNodeHandler) Handle(ctx context.Context, nCtx interfaces.NodeExecu
 
 		size := -1
 		for _, variable := range literalMap.Literals {
+			if variable.GetOffloadedMetadata() != nil {
+				// variable will be overwritten with the contents of the offloaded data which contains the actual large literal.
+				// We need this for the map task to be able to create the subNodeSpec
```

Comment:
just so I understand, this is also needed so the map task can index into the inputs as well too?

Reply:
yeah subNodeSpec covers that as it contains the inputs for each subnode